pytest/lib/mocknet.py

```diff
@@ -327,6 +336,103 @@ def stop_node(node):
         m.run('sudo -u ubuntu -i', input=TMUX_STOP_SCRIPT)
 
 
+def create_and_upload_genesis(nodes, genesis_template_filename):
+    mocknet_genesis_filename = '/tmp/mocknet_genesis.json'
+    create_genesis_file(nodes, genesis_template_filename, mocknet_genesis_filename)
+    # We assume that the nodes already have the .near directory with the files
+    # node_key.json, validator_key.json and config.json.
+    pmap(lambda node: node.machine.upload(mocknet_genesis_filename, '/home/ubuntu/.near/genesis.json'), nodes)
+    update_boot_nodes(nodes)
+
+
+def node_account_name(instance_name):
+    return f'{instance_name}.near'
+
+
+def create_genesis_file(nodes, genesis_template_filename, mocknet_genesis_filename):
+    with open(genesis_template_filename) as f:
+        MASTER_BALANCE = "1000000000000000000000000000000000"
+        TREASURY_BALANCE = MASTER_BALANCE
+        VALIDATOR_BALANCE = "20000000000000000000000000000000"
+        STAKED_BALANCE = "15000000000000000000000000000000"
+        MASTER_ACCOUNT = "near"
+        TREASURY_ACCOUNT = "test.near"
+
+        genesis_config = json.loads(f.read())
```

Comment:
with open(genisis_template_filename) as rd:
        genesis_config = json.load(rd)

and then dedent the rest of the code.